### PR TITLE
Update strings extract gbenchmark to measure multiple groups

### PR DESCRIPTION
**Pull request by davidwendt**
Reference https://github.com/rapidsai/custreamz/issues/2
Updating the gbenchmark for `cudf::strings::extract` to measure varying number of groups. Previously the regex pattern only had 1 group. This is to more closely align to the referenced issue which includes more than 20 groups.

### CODE REVIEW

File: cpp/benchmarks/string/extract_benchmark.cpp [hunk 1]
```diff
@@ -23,43 +25,54 @@
...
+  std::vector<std::string> samples(100);  // 100 unique rows of data to reuse
+  std::generate(samples.begin(), samples.end(), [&]() {
+    std::string row;  // build a row of random tokens
+    while (static_cast<int>(row.size()) < n_length)
+      row += std::to_string(words_dist(generator)) + " ";
+    return row;
+  });
+
+  std::string pattern;
+  while (static_cast<int>(pattern.size()) < groups) pattern += "(\\d+) ";
```
Comment (ttnghia) [inline cpp/benchmarks/string/extract_benchmark.cpp:50]:
```suggestion
  while (static_cast<int>(pattern.size()) < groups) { pattern += "(\\d+) "; }
```

File: cpp/benchmarks/string/extract_benchmark.cpp [hunk 2]
```diff
@@ -23,43 +25,54 @@
...
+  auto const n_length = static_cast<cudf::size_type>(state.range(1));
+
+  std::default_random_engine generator;
+  std::uniform_int_distribution<int> words_dist(0, 999);
+
+  std::vector<std::string> samples(100);  // 100 unique rows of data to reuse
+  std::generate(samples.begin(), samples.end(), [&]() {
+    std::string row;  // build a row of random tokens
+    while (static_cast<int>(row.size()) < n_length)
+      row += std::to_string(words_dist(generator)) + " ";
```
Comment (ttnghia) [inline cpp/benchmarks/string/extract_benchmark.cpp:45]:
```suggestion
    while (static_cast<int>(row.size()) < n_length) {
      row += std::to_string(words_dist(generator)) + " ";
    }
```

**Review by ttnghia — APPROVED** [review]

**Review by devavret — APPROVED** [review]

**Review by codereport — APPROVED** [review]

**Comment by ttnghia** [comment]
Rerun tests.

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**